### PR TITLE
Add fix in scoped_setups

### PR DESCRIPTION
**Pull request by JosseVanDelm**
Fix in scoped_setups, originally in #224 , but now cherry-picked from 9060797ccb8ef1ebe67078382ba1c60a5f5d569c

### CODE REVIEW

File: compiler/transforms/accfg_config_overlap.py [hunk 1]
```diff
@@ -171,6 +166,10 @@ def match_and_rewrite(self, op: accfg.SetupOp, rewriter: PatternRewriter, /):
             op.in_state.index - 1
         )  # -1 because the first block arg is the loop index
 
+        # also, if any operation between us and the loop start is not side effect free, abort
+        if any(not is_side_effect_free(prev_op) for prev_op in previous_ops_of(op)):
+            return
+
```
Comment (JosseVanDelm) [inline compiler/transforms/accfg_config_overlap.py:172]: So we can remove this now @AntonLydike

Comment (AntonLydike) [reply compiler/transforms/accfg_config_overlap.py:172]: Yes, it appears that we reached the conclusion that this was unneeded

**Comment by AntonLydike** [comment]
:muscle:

**Review by AntonLydike — APPROVED** [review]
:muscle:

**Pull request merged**